File: backend/src/Application/Auth/Roles.cs
namespace SponsorshipApproval.Application.Auth;

public static class Roles
{
    public const string Requestor = "Requestor";

    public const string Manager = "Manager";

    public const string FinanceAdmin = "FinanceAdmin";

    public const string SystemAdmin = "SystemAdmin";

    public static readonly IReadOnlyList<string> All =
    [
        Requestor,
        Manager,
        FinanceAdmin,
        SystemAdmin,
    ];
}

File: frontend/src/lib/roles.ts
export const Roles = {
  Requestor: 'Requestor',
  Manager: 'Manager',
  FinanceAdmin: 'FinanceAdmin',
  SystemAdmin: 'SystemAdmin',
} as const

export type Role = (typeof Roles)[keyof typeof Roles]

const roleLabels: Record<Role, string> = {
  [Roles.Requestor]: 'Requestor',
  [Roles.Manager]: 'Manager',
  [Roles.FinanceAdmin]: 'Finance Admin',
  [Roles.SystemAdmin]: 'System Admin',
}

export function getRoleLabel(role: Role): string {
  return roleLabels[role]
}

export function isRole(value: string): value is Role {
  return Object.values(Roles).includes(value as Role)
}
